ssv/committee.go

```diff
@@ -0,0 +1,187 @@
+package ssv
+
+import (
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+
+	spec "github.com/attestantio/go-eth2-client/spec/phase0"
+	"github.com/pkg/errors"
+	"github.com/ssvlabs/ssv-spec/qbft"
+	"github.com/ssvlabs/ssv-spec/types"
+)
+
+type CreateRunnerFn func(shareMap map[spec.ValidatorIndex]*types.Share) *CommitteeRunner
+
+type Committee struct {
+	Runners           map[spec.Slot]*CommitteeRunner
+	CommitteeMember   types.CommitteeMember
+	SignatureVerifier types.SignatureVerifier
+	CreateRunnerFn    CreateRunnerFn
+	Share             map[spec.ValidatorIndex]*types.Share
+}
+
+// NewCommittee creates a new cluster
+func NewCommittee(
+	committeeMember types.CommitteeMember,
+	verifier types.SignatureVerifier,
+	share map[spec.ValidatorIndex]*types.Share,
+	createRunnerFn CreateRunnerFn,
+) *Committee {
+	c := &Committee{
+		Runners:           make(map[spec.Slot]*CommitteeRunner),
+		CommitteeMember:   committeeMember,
+		SignatureVerifier: verifier,
+		CreateRunnerFn:    createRunnerFn,
+		Share:             share,
+	}
+	return c
+}
+
+// StartDuty starts a new duty for the given slot
+func (c *Committee) StartDuty(duty *types.CommitteeDuty) error {
+	if len(duty.BeaconDuties) == 0 {
+		return errors.New("no beacon duties")
+	}
+	if _, exists := c.Runners[duty.Slot]; exists {
+		return errors.New(fmt.Sprintf("CommitteeRunner for slot %d already exists", duty.Slot))
+	}
+	c.Runners[duty.Slot] = c.CreateRunnerFn(c.Share)
+	return c.Runners[duty.Slot].StartNewDuty(duty, c.CommitteeMember.GetQuorum())
+}
+
+// ProcessMessage processes Network Message of all types
+func (c *Committee) ProcessMessage(signedSSVMessage *types.SignedSSVMessage) error {
+	// Validate message
+	if err := signedSSVMessage.Validate(); err != nil {
+		return errors.Wrap(err, "invalid SignedSSVMessage")
+	}
+
+	// Verify SignedSSVMessage's signature
+	if err := c.SignatureVerifier.Verify(signedSSVMessage, c.CommitteeMember.Committee); err != nil {
+		return errors.Wrap(err, "SignedSSVMessage has an invalid signature")
+	}
+
+	msg := signedSSVMessage.SSVMessage
+	if err := c.validateMessage(msg); err != nil {
+		return errors.Wrap(err, "Message invalid")
+	}
+
+	switch msg.GetType() {
```

Comment:
Also if we look at your solution we have pretty much the same duplicate code

https://github.com/ssvlabs/ssv-spec/compare/main...alonmuroch:ssv-spec:sip42#diff-2fc0f1ec524fe4c13facd7998fa8fc4d2738cdda55f95e30c194bbaac2894f71R74-R99

The only solution is to completely remove `Validator` and have everything committee based

Reply:
I agree next step is to merge "validator" into committee handle all the duties within the committee. it could be as easy as adding the validator runners inside the committee struct. anyway its not a fork so we can do it in next versions.